src/ddb/ddb.h

```diff
@@ -5,7 +5,92 @@
  */
 #ifndef __DDB_RUN_CMDS_H
 #define __DDB_RUN_CMDS_H
-#include "ddb_common.h"
+
+
+#include <daos_types.h>
+
+typedef int (*ddb_io_line_cb)(void *cb_args, char *line, uint32_t str_len);
+
+struct ddb_io_ft {
+	/**
+	 * Print a message.
+	 *
+	 * @param fmt	Typically printf string format
+	 * @param ...	Additional args will be formatted into the printed string
+	 * @return	Total number of characters written
+	 */
+	int (*ddb_print_message)(const char *fmt, ...);
+
+	/**
+	 * Print an error message.
+	 *
+	 * @param fmt	Typically printf string format
+	 * @param ...	Additional args will be formatted into the printed string
+	 * @return	Total number of characters written
+	 */
+	int (*ddb_print_error)(const char *fmt, ...);
+
+	/**
+	 * Read a line from stdin and stores into buf.
+	 *
+	 * @param buf		Pointer to an array where the string read is stored
+	 * @param buf_len	Length of buf
+	 * @return		On success the same buf parameter, else NULL
+	 */
+	char *(*ddb_get_input)(char *buf, uint32_t buf_len);
+
+	/**
+	 * Check if a file exists
+	 *
+	 * @param path	Path to file to check
+	 * @return	true if the file exists, else false
+	 */
+	bool (*ddb_get_file_exists)(const char *path);
+
+	/**
+	 * Write the contents of the iov to a file
+	 *
+	 * @param dst_path	File to write to
+	 * @param contents	Contents to be written
+	 * @return		0 on success, else an error code
+	 */
+	int (*ddb_write_file)(const char *dst_path, d_iov_t *contents);
+
+	/**
+	 * Determine the size of a file at path
+	 * @param path	Path of file to check
+	 * @return	the size of the file at path in bytes
+	 */
+	size_t (*ddb_get_file_size)(const char *path);
+
+	/**
+	 * Read the contents of a file and store into the iov
+	 * @param src_path	Path of the file to read
+	 * @param contents	Where to load the contents of the file into
+	 * @return		number of bytes read from the src_path
+	 */
+	size_t (*ddb_read_file)(const char *src_path, d_iov_t *contents);
+
+	/**
+	 * Read contents of a file line by line. For each line, the line_cb will be called.
+	 * @param path		Path of the file to read
+	 * @param line_cb	Callback function used for each line
+	 * @param cb_args	Caller arguments passed to the callback function
+	 * @return		0 on success, else an error code
+	 */
+	int (*ddb_get_lines)(const char *path, ddb_io_line_cb line_cb, void *cb_args);
+};
+
+struct ddb_ctx {
+	struct ddb_io_ft	 dc_io_ft;
+	daos_handle_t		 dc_poh;
+	bool			 dc_should_quit;
+	bool			 dc_write_mode;
+};
+
+void ddb_ctx_init(struct ddb_ctx *ctx);
+int ddb_init();
+void ddb_fini();
```

Comment:
(style)  Bad function definition - void ddb_fini() should probably be void ddb_fini(void)